pkg/authn/keychain_test.go

```diff
@@ -363,3 +365,27 @@ func TestNewKeychainFromHelper(t *testing.T) {
 		}
 	})
 }
+
+func TestConfigFileIsADir(t *testing.T) {
+	tmpdir := setupConfigDir(t)
+	// Create "config.json" as a directory, not a file to simulate optional
+	// secrets in Kubernetes.
+	err := os.Mkdir(path.Join(tmpdir, "config.json"), 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Clear out files that may have been created by previous tests.
+	err = os.RemoveAll(path.Join(os.Getenv("HOME"), ".docker"))
```

Comment:
Is this going to delete my ~/.docker directory if I run this test? 🤔

Reply:
It shouldn't because `TestMain` sets `HOME` to a temporary location. I agree that it does look odd...

Let me see if I can refactor how `HOME` is set in the first place.

Reply:
That's okay, let's keep it how it is unless we want to fix it in a separate PR. Thanks for clarifying how it works.

Reply:
Added a defer call to remove the `.docker/config.json` right after it is created in the corresponding test.